flink-streaming-connectors/flink-connector-redis/src/main/java/org/apache/flink/streaming/connectors/redis/RedisSink.java

```diff
@@ -0,0 +1,224 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.flink.streaming.connectors.redis;
+
+import org.apache.flink.configuration.Configuration;
+import org.apache.flink.streaming.api.functions.sink.RichSinkFunction;
+import org.apache.flink.streaming.connectors.redis.common.config.JedisClusterConfig;
+import org.apache.flink.streaming.connectors.redis.common.config.JedisPoolConfig;
+import org.apache.flink.streaming.connectors.redis.common.config.JedisSentinelConfig;
+import org.apache.flink.streaming.connectors.redis.common.container.RedisCommandsContainer;
+import org.apache.flink.streaming.connectors.redis.common.container.RedisCommandsContainerBuilder;
+import org.apache.flink.streaming.connectors.redis.common.mapper.RedisDataType;
+import org.apache.flink.streaming.connectors.redis.common.mapper.RedisDataTypeDescription;
+import org.apache.flink.streaming.connectors.redis.common.mapper.RedisMapper;
+
+import redis.clients.jedis.exceptions.JedisException;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.base.Preconditions;
+
+/**
+ * A sink that delivers data to a Redis channel using the Jedis client.
+ *
+ *  <p>
+ * When creating the sink using first constructor {@link #RedisSink(JedisPoolConfig, RedisMapper)}
+ * the sink will create connection using {@link redis.clients.jedis.JedisPool}.
+ *
+ * When using second constructor {@link #RedisSink(JedisSentinelConfig, RedisMapper)} the sink will create connection
+ * using {@link redis.clients.jedis.JedisSentinelPool} to redis cluster. Use this if redis is
+ * configured using sentinels else use the third constructor {@link #RedisSink(JedisClusterConfig, RedisMapper)}
+ * which use {@link redis.clients.jedis.JedisCluster} to connect to redis cluster.
+ *
+ * <p>
+ * Example:
+ *
+ * <pre>{@code
+ *
+ * 		public static class RedisExampleDataMapper implements RedisMapper<Tuple2<String, String>>{
+ *			@Override
+ *			public RedisDataTypeDescription getDataTypeDescription() {
+ *				return new RedisDataTypeDescription(dataType, REDIS_ADDITIONAL_KEY);
+ *			}
+ *			@Override
+ *			public String getKeyFromData(Tuple2 data) {
+ *				return String.valueOf(data.f0);
+ *			}
+ *
+ *			@Override
+ *			public String getValueFromData(Tuple2 data) {
+ *				return String.valueOf(data.f1);
+ *			}
+ *		}
+ *		JedisPoolConfig jedisPoolConfig = new JedisPoolConfig.Builder()
+ *		.setHost(REDIS_HOST)
+ *		.setPort(REDIS_PORT).build();
+ *      new RedisSink<String>(jedisPoolConfig, new RedisExampleDataMapper());
+ * }</pre>
+ *
+ * @param <IN> Type of the elements emitted by this sink
+ */
+public class RedisSink<IN> extends RichSinkFunction<IN>{
+
+	private static final long serialVersionUID = 1L;
+
+	private static final Logger LOG = LoggerFactory.getLogger(RedisSink.class);
+
+	private String additionalKey;
```

Comment:
For me as an redis newby, it is unclear was `additionalKey` might be. Might be good to add a comment with a short explanation/description.